util/codec/codec.go

```diff
@@ -172,3 +172,145 @@ func DecodeKey(b []byte) ([]interface{}, error) {
 
 	return values, nil
 }
+
+// EncodeValue encodes args to a byte slice which can be decoded by DecodeValue later.
+// It does not guarantee the order for comparison.
+func EncodeValue(args ...interface{}) ([]byte, error) {
```

Comment:
Got it.
BTW, how about adding new flags for varint and compactBytes? Then we can use a common decode function.